### PR TITLE
Improve crossgen2 comparison jobs

### PR DESCRIPTION
**Pull request by davidwrighton**
- Fix compilation on unix platforms
  - Wrap use of wildcard in quotes
- Print better display name into log
- Fix X86 constant comparison handling
- Add ability to compile specific overload via single method switches

### CODE REVIEW

**Comment by davidwrighton** [comment]
@dotnet/jit-contrib I have 1 small change to the JIT here which addresses a missed optimization triggered by unexpected sign-extension.

@dotnet/crossgen-contrib This change brings us ever closer to crossgen2 comparison jobs passing. With this change, there is 1 file which failed in X64 to Arm64 compilation comparison, which I wasn't able to repro locally. We should get this state checked in and then do further testing. Also, I added a mechanism for using single method with overloads as I needed that during investigation of various issues here

File: eng/pipelines/coreclr/templates/crossgen2-comparison-job.yml [hunk 1]
```diff
@@ -154,6 +157,7 @@ jobs:
             --diff_dir   $HELIX_WORKITEM_PAYLOAD/log
         ${{ if eq(parameters.osGroup, 'windows') }}:
           WorkItemCommand:
+            echo $(displayname_comparison_job)
```
Comment (BruceForstall) [inline eng/pipelines/coreclr/templates/crossgen2-comparison-job.yml:160]: Not sure how this is set up, but does this line need a trailing `&` like the commands that follow? (Looks like this is set to create one long single-line batch script?)

**Review by BruceForstall — APPROVED** [review]
LGTM (didn't look at crossgen2 source)

Presumably you're planning to add more cross-comparison jobs later, like arm32 cross versus arm32 native, x64->x86 cross versus x86 native, etc.?

File: eng/pipelines/coreclr/templates/crossgen2-comparison-job.yml (hunk 1 above)
Comment (davidwrighton) [reply eng/pipelines/coreclr/templates/crossgen2-comparison-job.yml:160]: It should have the &, but it does work. See https://helixre8s23ayyeko0k025g8.blob.core.windows.net/dotnet-runtime-refs-pull-44119-head-8c5be3842f80486d85/WorkItem/console.9e397036.log

**Review by nattress — APPROVED** [review]
LGTM - Thanks!

**Pull request merged**